rs/replicated_state/src/canister_state/queues/input_schedule/tests.rs

```diff
@@ -0,0 +1,319 @@
+use super::*;
+use crate::canister_state::queues::{
+    message_pool::MessagePool, tests::input_queue_type_from_local_canisters,
+};
+use assert_matches::assert_matches;
+use ic_test_utilities_types::messages::RequestBuilder;
+use InputQueueType::*;
+
+#[test]
+fn test_input_source() {
+    let mut schedule = InputSchedule::default();
+
+    assert_eq!(InputSource::default(), schedule.input_source());
+    assert_eq!(InputSource::default(), schedule.next_input_source());
+
+    assert_eq!(InputSource::Ingress, schedule.input_source());
+    assert_eq!(InputSource::Ingress, schedule.next_input_source());
+
+    assert_eq!(InputSource::RemoteSubnet, schedule.input_source());
+    assert_eq!(InputSource::RemoteSubnet, schedule.next_input_source());
+
+    assert_eq!(InputSchedule::default(), schedule);
+
+    assert_eq!(InputSource::LocalSubnet, schedule.input_source());
+    assert_eq!(InputSource::LocalSubnet, schedule.next_input_source());
```

Comment:
We just tested that it circled back to `LocalSubnet` (`InputSource::default()` is `LocalSubnet`; which is also why we're checking that the schedule as a whole is equal to `InputSchedule::default()`).

We would just be doing N+2 iterations instead of N+1.